pkg/mimirtool/analyze/grafana.go

```diff
@@ -198,13 +198,26 @@ func metricsFromPanel(panel minisdk.Panel, metrics map[string]struct{}) []error
 }
 
 func parseQuery(query string, metrics map[string]struct{}) error {
+	// replace standard Grafana Prometheus macros
 	query = strings.ReplaceAll(query, `$__interval`, "5m")
 	query = strings.ReplaceAll(query, `$interval`, "5m")
 	query = strings.ReplaceAll(query, `$resolution`, "5s")
 	query = strings.ReplaceAll(query, "$__rate_interval", "15s")
 	query = strings.ReplaceAll(query, "$__range", "1d")
 	query = strings.ReplaceAll(query, "${__range_s:glob}", "30")
 	query = strings.ReplaceAll(query, "${__range_s}", "30")
+	// replace duration variable, e.g. [$agregation_window]
+	re := regexp.MustCompile(`\[\$.+\]`)
```

Comment:
We should be able to move these `regexp.MustCompile` calls out of `parseQuery` - otherwise they'll be recompiled for every `parseQuery` call, which is potentially expensive.